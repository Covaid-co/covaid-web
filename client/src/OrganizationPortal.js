import React, { useState, useEffect } from "react";
import { useToasts } from "react-toast-notifications";
import fetch_a from "./util/fetch_auth";
import Pusher from "pusher-js";

import PropTypes from "prop-types";

import Container from "react-bootstrap/Container";
import Jumbotron from "react-bootstrap/Jumbotron";
import Button from "react-bootstrap/Button";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Badge from "react-bootstrap/Badge";
import OrgLogin from "./components_orgpage/OrgLogin";
import Cookie from "js-cookie";
import NavBar from "./components/NavBar";
import Footer from "./components/Footer";

import RequestDetails from "./components_orgpage/RequestDetails";
import VolunteerDetails from "./components_orgpage/VolunteerDetails";
import VolunteersModal from "./components_orgpage/VolunteersModal";
import AdminModal from "./components_orgpage/AdminModal";
import BeaconCreation from "./components_orgpage/BeaconCreation";
import OrgResourcesModal from "./components_orgpage/OrgResourcesModal";
import LiveBeaconView from "./components_orgpage/LiveBeaconView";
import OrganizationRequestBoard from "./components_orgpage/OrganizationRequestBoard";
import OrganizationMap from "./components_orgpage/OrganizationMap";
import {
  fetchBeacons,
  fetchOrgVolunteers,
  getName,
  fetchOrgRequests,
} from "./components_orgpage/OrganizationHelpers";
import "./OrganizationPage.css";

export default function OrganiationPortal(props) {
  const { addToast } = useToasts();
  const [pageLoaded, setPageLoaded] = useState(false);
  const [currTabNumber, setCurrTab] = useState(0);
  const [showLogin, setShowLogin] = useState(false);
  const [association, setAssociation] = useState({});
  const [volunteers, setVolunteers] = useState([]);
  const [volunteersModal, setVolunteersModal] = useState(false);
  const [adminModal, setAdminModal] = useState(false);
  const [beaconModal, setBeaconModal] = useState(false);
  const [resourceModal, setResourceModal] = useState(false);
  const [allRequests, setAllRequests] = useState([]);
  const [beacons, setBeacons] = useState([]);
  const [volunteerDetailModal, setVolunteerDetailsModal] = useState(false);
  const [requestDetailsModal, setRequestDetailsModal] = useState(false);
  const [currVolunteer, setCurrVolunteer] = useState({});
  const [currRequest, setCurrRequest] = useState({
    admin_info: {
      assignee: "1",
    },
    status: {
      current_status: 1,
      volunteers: [],
    },
    personal_info: {
      requester_name: "1",
      requester_email: "1",
      languages: ["English"],
    },
    request_info: {
      resource_request: ["1"],
      payment: 1,
      details: "1",
      time: "1",
      date: "1",
    },
    association: "5e88cf8a6ea53ef574d1b80c",
    location_info: { coordinates: [1, 1] },
  });

  const [admin, setAdmin] = useState({});

  const [beaconView, setBeaconView] = useState(false);
  const [width, setWidth] = useState(window.innerWidth);

  const [inRequest, setInRequest] = useState(false);
  const [inVolunteer, setInVolunteer] = useState(false);
  //console.log(props.switchToLanguage);
  useEffect(() => {
    if (Cookie.get("admin_token") && Cookie.get("org_token")) {
      login(true);
    } else if (Cookie.get("org_token")) {
      login(false);
    } else {
      setShowLogin(true);
    }
  }, []);

  window.addEventListener("resize", () => {
    setWidth(window.innerWidth);
  });

  const pushBeacon = (beacon) => {
    setBeacons(beacons.concat(beacon));
  };

  function fetchCurrentAdmin() {
    fetch_a("admin_token", "/api/association-admin/current")
      .then((response) => response.json())
      .then((adminResponse) => {
        setAdmin(adminResponse);
        setPageLoaded(true);
      })
      .catch((e) => {
        alert(e);
      });
  }

  const fetch_requests = (id) => {
    fetchOrgRequests(id).then((requests) => {
      setAllRequests(requests);
    });
  };

  const pusherSetup = (id) => {
    var pusher = new Pusher("ed72954a8d404950e3c8", {
      cluster: "us2",
      forceTLS: true,
    });
    var channel = pusher.subscribe(id);
    channel.bind("general", function (data) {
      fetch_requests(id);
      addToast(data, {
        appearance: "info",
        autoDismiss: true,
      });
    });
    channel.bind("complete", function () {
      fetch_requests(id);
      addToast("Someone completed a request!", {
        appearance: "success",
        autoDismiss: true,
      });
    });
    channel.bind("request-details", function () {
      fetch_requests(id);
    });
  };

  function login(adminMode) {
    // Get association from login
    fetch('/api/association/current_demo')
      .then((response) => response.json())
      .then((association_response) => {
        setAssociation(association_response);
        pusherSetup(association_response._id);

        // Get requests for an association
        fetch_requests(association_response._id);

        // All beacons
        fetchBeacons().then((beacons) => {
          setBeacons(beacons);
        });

        // Get all volunteers for an association
        fetchOrgVolunteers(association_response._id).then((volunteers) => {
          setVolunteers(volunteers);
        });

        if (adminMode) {
          fetchCurrentAdmin();
        } else {
          setPageLoaded(true);
        }
      })
      .catch((e) => {
        alert(e);
      });
  }

  if (beaconView) {
    return (
      <LiveBeaconView
        volunteers={volunteers}
        association={association}
        setBeaconView={setBeaconView}
        beacons={beacons}
      />
    );
  } else if (showLogin === true) {
    return (
      <OrgLogin
        setLanguage={props.setLanguage}
        language={props.language}
        login={login}
        setShowLogin={setShowLogin}
        orgReset={props.location.orgReset}
      />
    );
  } else if (!pageLoaded) {
    return <></>;
  }
  return [
    <div className="App" key="1">
      <NavBar
        setLanguage={props.setLanguage}
        language={'en'}
        pageLoaded={pageLoaded}
        isLoggedIn={true}
        totalVolunteers={volunteers.length}
        setAdmin={setAdmin}
        orgPortal={true}
        first_name={getName(admin, association)}
        handleShowModal={() => {}}
      />
      <div style={{ zoom: "95%" }}>
        <Jumbotron
          fluid
          id="jumbo-volunteer"
          style={{ paddingBottom: 50, paddingTop: 60 }}
        >
          <Container style={{ maxWidth: 2000 }}>
            <Row>
              <Col lg={6} md={6} sm={12}>
                <h1 id="home-heading" style={{ marginTop: 0 }}>
                  Welcome back,
                </h1>
                <h1 id="home-heading" style={{ marginTop: 0 }}>
                  {association.name}!
                </h1>
                <p id="regular-text" style={{ fontSize: 20, marginBottom: 40 }}>
                  This is your organization portal, a place for you to manage
                  volunteers and requests in your area
                </p>
                <Button
                  id="medium-button"
                  style={{ marginRight: 10, marginTop: 5 }}
                  onClick={() => {
                    setAdminModal(true);
                  }}
                >
                  Manage Organization
                </Button>
                <Button
                  id="medium-button"
                  style={{ marginTop: 5 }}
                  onClick={() => {
                    setVolunteersModal(true);
                  }}
                >
                  View Volunteers
                </Button>{" "}
                <br />
                <Button
                  variant="link"
                  id="resources-link"
                  onClick={() => {
                    setResourceModal(true);
                  }}
                >
                  + Add a link to your community&apos;s resources
                </Button>
              </Col>
              <Col
                lg={6}
                md={6}
                sm={12}
                style={width < 768 ? { display: "none" } : { display: "block" }}
              >
                <Container
                  id="newOfferContainer"
                  style={{
                    width: "75%",
                    marginBottom: 0,
                    position: "absolute",
                    marginTop: 20,
                  }}
                >
                  <h3 id="home-heading" style={{ marginTop: 0, fontSize: 20 }}>
                    Need a task done?{" "}
                    <Badge id="task-info" style={{ background: "#AE2F2F" }}>
                      BETA
                    </Badge>
                  </h3>
                  <p id="regular-text" style={{ marginBottom: 10 }}>
                    Use our <b>Beacon Notification System</b> and mass notify
                    your volunteers about any internal organization requests
                  </p>
                  <Row>
                    <Col style={{ paddingRight: 5 }}>
                      <Button
                        id="large-button"
                        onClick={() => {
                          setBeaconModal(true);
                        }}
                      >
                        Create Beacon
                      </Button>
                    </Col>
                    <Col style={{ paddingLeft: 5 }}>
                      <Button
                        id="large-button-empty"
                        style={{ marginTop: 0, paddingLeft: 5 }}
                        onClick={() => {
                          setBeaconView(true);
                        }}
                      >
                        Live Beacons (
                        {
                          beacons.filter((beacon) => beacon.beaconStatus === 1)
                            .length
                        }
                        )
                      </Button>
                    </Col>
                  </Row>
                </Container>
              </Col>
            </Row>
          </Container>
        </Jumbotron>
        <Container style={{ maxWidth: 2000 }}>
          <Row className="justify-content-md-center">
            <Col lg={6} md={12} sm={12} style={{ marginTop: -44 }}>
              <OrganizationRequestBoard
                currTabNumber={currTabNumber}
                setCurrTab={setCurrTab}
                allRequests={allRequests}
                setInRequest={setInRequest}
                setCurrRequest={setCurrRequest}
                setRequestDetailsModal={setRequestDetailsModal}
              />
            </Col>
            <Col lg={6} md={12} sm={12}>
              <OrganizationMap
                requests={allRequests}
                volunteers={volunteers}
                mode={currTabNumber}
                allRequests={allRequests}
                volunteerDetailModal={volunteerDetailModal}
                association={association}
                setVolunteerDetailsModal={setVolunteerDetailsModal}
                width={width}
                currVolunteer={currVolunteer}
                setCurrVolunteer={setCurrVolunteer}
                requestDetailsModal={requestDetailsModal}
                setRequestDetailsModal={setRequestDetailsModal}
                currRequest={currRequest}
                setCurrRequest={setCurrRequest}
                setInRequest={setInRequest}
              />
            </Col>
          </Row>
        </Container>
        <VolunteersModal
          volunteersModal={volunteersModal}
          setVolunteersModal={setVolunteersModal}
          volunteers={volunteers}
          association={association}
          setCurrVolunteer={setCurrVolunteer}
          setVolunteerDetailsModal={setVolunteerDetailsModal}
          setInVolunteer={setInVolunteer}
        />
        <AdminModal
          adminModal={adminModal}
          setAdminModal={setAdminModal}
          association={association}
          setAssociation={setAssociation}
        />
        <OrgResourcesModal
          resourceModal={resourceModal}
          setResourceModal={setResourceModal}
          association={association}
          setAssociation={setAssociation}
        />
        <VolunteerDetails
          volunteerDetailModal={volunteerDetailModal}
          setVolunteerDetailsModal={setVolunteerDetailsModal}
          currVolunteer={currVolunteer}
          setVolunteersModal={setVolunteersModal}
          currRequest={currRequest}
          requestDetailsModal={requestDetailsModal}
          setRequestDetailsModal={setRequestDetailsModal}
          inRequest={inRequest}
          inVolunteer={inVolunteer}
        />
        <RequestDetails
          requestDetailsModal={requestDetailsModal}
          setRequestDetailsModal={setRequestDetailsModal}
          volunteerDetailModal={volunteerDetailModal}
          setVolunteerDetailsModal={setVolunteerDetailsModal}
          setCurrVolunteer={setCurrVolunteer}
          currRequest={currRequest}
          setCurrRequest={setCurrRequest}
          association={association}
          allRequests={allRequests}
          setAllRequests={setAllRequests}
          mode={currTabNumber}
          volunteers={volunteers}
          admin={admin}
          setInRequest={setInRequest}
        />
        <BeaconCreation
          beaconModal={beaconModal}
          setBeaconModal={setBeaconModal}
          association={association}
          volunteers={volunteers}
          pushBeacon={pushBeacon}
          switchToBeacon={() => setBeaconView(true)}
        />
      </div>
    </div>,
    <Footer key="2" />,
  ];
}

OrganiationPortal.propTypes = {
  language: PropTypes.string,
  setLanguage: PropTypes.func,
};
