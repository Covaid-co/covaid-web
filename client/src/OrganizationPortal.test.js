import React from "react";
import ReactDOM from 'react-dom';
import { render } from "@testing-library/react";
import {shallow, mount} from 'enzyme';
import OrganizationPortal from "./OrganizationPortal";
import { ToastProvider } from "react-toast-notifications";
import { act } from 'react-dom/test-utils';

function OrganizationPortalPage(props) {
	return (
		<ToastProvider>
			<OrganizationPortal {...props}/>
		</ToastProvider>
	);
}

describe ('<OrganizationPortal />', () => {
	const props = {
	    location: {
	    	orgReset: true,
	    },
	};

	const container = mount(<OrganizationPortalPage {...props}/>);
	it('should match the snapshot', () => {
		expect(container.html()).toMatchSnapshot();
	});

	describe('Organization Login <OrgLogin />', () => {
		it('should have proper props for email field', () => {

	    	expect(container.find('input[type="email"]').find('input[id="emailOrg"]').props()).toEqual({
	      		onChange: expect.any(Function),
	      		placeholder: 'Email',
	      		type: 'email',
	      		className: "form-control",
    	   		id: "emailOrg",
    	   		readOnly: undefined,
    	  		value: "",
		    });
		});

		it('should have proper props for password field', () => {
		    expect(container.find('input[type="password"]').props()).toEqual({
		      	placeholder: 'Password',
		      	onChange: expect.any(Function),
		      	type: "password",
		      	readOnly: undefined,
		      	className: "form-control",
       			id: "passOrg",
       			value: "",
		    });
		});

		// it('should set the email value on change event', () => {
		// 	beforeAll(() => {
		// 	    container.find('input[id="emailOrg"]').prop('onChange')({
		// 	      	target: {
		// 	      		id: 'emailOrg',
		// 	        	value: 'somenewemail@gmail.com',
		// 	      	},
		// 	    })
		// 	})
		//     expect(container.find('input[id="emailOrg"]').prop('value')).toEqual(
		//       	'somenewemail@gmail.com',
		//     );
		// });

		// it('should set the password value on change event', () => {
		//     container.find('input[type="password"]').prop('onChange')({
		//       	target: {
		//       		id: 'passOrg',
		//         	value: 'somenewpassword',
		//       	},
		//     });
		//     expect(container.find('input[type="password"]').prop('value')).toEqual(
		//       	'somenewpassword',
		//     );
		// });

	});


});



