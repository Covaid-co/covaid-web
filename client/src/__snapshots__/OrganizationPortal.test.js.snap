// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrganizationPortal /> should match the snapshot 1`] = `"<div class=\\"App\\"><nav id=\\"custom-navbar\\" class=\\"navbar navbar-expand-md navbar-light\\"><a href=\\"http://localhost\\" id=\\"navbar-brand\\" class=\\"navbar-brand\\">covaid</a><form class=\\"volunteer-badge-mobile form-inline\\"><button aria-controls=\\"basic-navbar-nav\\" id=\\"nav1\\" type=\\"button\\" aria-label=\\"Toggle navigation\\" class=\\"navbar-toggler collapsed\\"><span class=\\"navbar-toggler-icon\\"></span></button></form><div class=\\"navbar-collapse collapse\\" id=\\"basic-navbar-nav\\"><div class=\\"mr-auto navbar-nav\\"><a href=\\"#\\" class=\\"navbar-element nav-link\\" role=\\"button\\"><p id=\\"navLink\\">About us</p></a><a href=\\"http://localhost/organizationPortal\\" data-rb-event-key=\\"http://localhost/organizationPortal\\" class=\\"navbar-border-element nav-link\\"><p id=\\"navLinkBorder\\">Organizations</p></a><a href=\\"#\\" class=\\"navbar-element nav-link\\" role=\\"button\\"><p id=\\"navLink\\">FAQs</p></a><a href=\\"#\\" class=\\"navbar-element nav-link\\" role=\\"button\\"><p id=\\"navLink\\">Donate</p></a><a href=\\"#\\" class=\\"volunteer-badge-web nav-link\\" role=\\"button\\"></a></div></div></nav><div style=\\"max-width: 1500px;\\" class=\\"container\\"><div class=\\"row\\"><div id=\\"login-container\\" class=\\"col-md-6\\"><h1 id=\\"home-heading\\">Covaid for Organizations</h1><p id=\\"home-subheading\\" style=\\"font-size: 16px;\\">Manage and delegate your volunteers efficiently through our all-in-one platform.</p><form class=\\"\\"><div class=\\"row\\"><div class=\\"col-12\\"><div bssize=\\"large\\" style=\\"margin-bottom: 5px;\\" class=\\"form-group\\"><input placeholder=\\"Email\\" type=\\"email\\" id=\\"emailOrg\\" class=\\"form-control\\" value=\\"\\"></div></div><div class=\\"col-12\\"><div bssize=\\"large\\" class=\\"form-group\\"><input placeholder=\\"Password\\" type=\\"password\\" id=\\"passOrg\\" class=\\"form-control\\" value=\\"\\"></div></div></div><button style=\\"margin-top: 10px; width: 150px;\\" id=\\"large-button\\" disabled=\\"\\" type=\\"submit\\" class=\\"btn btn-primary\\">Sign In</button><p id=\\"regular-text\\" style=\\"margin-top: 15px; color: rgb(38, 112, 255);\\">Manage a mutual aid initiative?<button id=\\"regular-text\\" style=\\"color: rgb(38, 112, 255); padding: 0px; text-decoration: underline; margin-top: -2px; margin-left: 5px;\\" type=\\"button\\" class=\\"btn btn-link\\">Get Started</button></p></form></div><div style=\\"margin-top: 50px;\\" class=\\"col-md-6\\"><img id=\\"org-img\\" alt=\\"\\" src=\\"orgNew.png\\"></div></div><div id=\\"orgpage-separator\\" class=\\"row\\"><div class=\\"col-12\\"><p id=\\"requestCall\\" style=\\"margin-top: 30px; margin-bottom: 0px;\\">&nbsp;</p></div></div><div style=\\"margin-top: 80px; margin-bottom: 100px; padding-left: 10%; padding-right: 10%;\\" class=\\"row\\"><div id=\\"org-feature-container\\" class=\\"col-12\\"><h1 id=\\"home-sub-heading\\">Our Current Partners</h1></div><div id=\\"partner\\" class=\\"col-md-4\\"><div class=\\"inner-partner\\" style=\\"background-image: url(pitt.png); background-size: cover; background-position: center;\\"><h1 id=\\"partner-heading\\">Pittsburgh Mutual Aid</h1></div></div><div id=\\"partner\\" class=\\"col-md-4\\"><div class=\\"inner-partner\\" style=\\"background-image: url(bmore.png); background-size: cover; background-position: center;\\"><h1 id=\\"partner-heading\\">Baltimore Mutual Aid</h1></div></div><div id=\\"partner\\" class=\\"col-md-4\\"><div class=\\"inner-partner\\" style=\\"background-image: url(char.png); background-size: cover; background-position: center;\\"><h1 id=\\"partner-heading\\">Greater Charlotte Area Mutual Aid</h1></div></div><div id=\\"partner\\" class=\\"col-md-4\\"><div class=\\"inner-partner\\" style=\\"background-image: url(delaware.png); background-size: cover; background-position: center;\\"><h1 id=\\"partner-heading\\">Delaware Mutual Aid</h1></div></div><div id=\\"partner\\" class=\\"col-md-4\\"><div class=\\"inner-partner\\" style=\\"background-image: url(ccom.png); background-size: cover; background-position: center;\\"><h1 id=\\"partner-heading\\">CCOM COVID-19 Task Force</h1></div></div><div id=\\"partner\\" class=\\"col-md-4\\"><div class=\\"inner-partner\\"><h1 id=\\"partner-heading\\">Indy Neighbor Response Team</h1></div></div></div><div style=\\"margin-top: 15px;\\" class=\\"row\\"><div id=\\"requests-feature-mobile\\" class=\\"col-md-6\\"><h1 id=\\"home-sub-heading\\">Track Requests</h1><p id=\\"home-subheading\\" style=\\"font-size: 16px; padding-right: 0px; margin-bottom: 0px;\\">Optimize your workflow with the Covaid request tracker, an automated system for managing resource requests coming to your organization.</p></div><div id=\\"requests-feature-container\\" style=\\"text-align: center;\\" class=\\"col-md-6\\"><img id=\\"request-img\\" alt=\\"\\" src=\\"requests.png\\"></div><div id=\\"feature-container\\" class=\\"col-md-6\\"><h1 id=\\"home-sub-heading\\">Track Requests</h1><p id=\\"home-subheading\\" style=\\"font-size: 16px;\\">Optimize your workflow with the Covaid Request Tracker, an automated system for managing resource requests coming to your organization.</p></div></div><div style=\\"margin-top: 45px;\\" class=\\"row\\"><div id=\\"location-feature-container\\" class=\\"col-md-6\\"><h1 id=\\"home-sub-heading\\">Location Tracker</h1><p id=\\"home-subheading\\" style=\\"font-size: 16px; padding-right: 0px; margin-bottom: 0px;\\">Delegate requests by location and better understand your volunteer base. The map integration also allows organization leaders to view the general location of in-progress and completed requests.</p></div><div id=\\"map-feature-container\\" class=\\"col-md-6\\"><img id=\\"request-img\\" alt=\\"\\" src=\\"mapscreen.png\\"></div></div><div style=\\"margin-top: 45px;\\" class=\\"row\\"><div id=\\"requests-feature-mobile\\" class=\\"col-md-6\\"><h1 id=\\"home-sub-heading\\">Match Volunteers</h1><p id=\\"home-subheading\\" style=\\"font-size: 16px; padding-right: 0px; margin-bottom: 0px;\\">Easily match and notify volunteers in your organization to requests based on the recommended volunteers we offer.</p></div><div id=\\"requests-feature-container\\" style=\\"text-align: center;\\" class=\\"col-md-6\\"><img id=\\"request-img\\" alt=\\"\\" src=\\"topmatches.png\\"></div><div id=\\"feature-container\\" class=\\"col-md-6\\"><h1 id=\\"home-sub-heading\\" style=\\"margin-top: 140px;\\">Match Volunteers</h1><p id=\\"home-subheading\\" style=\\"font-size: 16px;\\">Easily match and notify volunteers in your organization to requests based on the recommended volunteers we offer.</p></div></div><div style=\\"margin-top: 80px; margin-bottom: 100px;\\" class=\\"row\\"><div id=\\"org-feature-container\\" class=\\"col-12\\"><h1 id=\\"home-sub-heading\\">Take your next step</h1><p id=\\"home-subheading\\" style=\\"font-size: 20px; padding-right: 0px; margin-bottom: 0px;\\">Weâ€™re excited to work with you and help grow your efforts!</p><button style=\\"margin-top: 10px; width: 250px;\\" id=\\"large-button\\" type=\\"button\\" class=\\"btn btn-primary\\">Get Started</button></div></div></div><footer class=\\"footer\\"><div class=\\"container\\"><div style=\\"text-align: left; padding-bottom: 30px;\\" class=\\"row\\"><div style=\\"padding: 20px 0px 20px 20px;\\" class=\\"col-2\\"><nav style=\\"padding-left: 0px; padding-bottom: 0px;\\" class=\\"navbar navbar-expand navbar-light\\"><span id=\\"navbar-brand\\" style=\\"padding-left: 0px; color: rgb(123, 123, 123); margin-left: 0px;\\" class=\\"navbar-brand\\">covaid</span></nav><button id=\\"regular-text\\" style=\\"color: rgb(38, 112, 255); padding: 0px;\\" type=\\"button\\" class=\\"btn btn-link\\">FAQ</button><br><button id=\\"regular-text\\" style=\\"color: rgb(38, 112, 255); padding: 0px; margin-top: -8px;\\" type=\\"button\\" class=\\"btn btn-link\\">Updates</button><br><button id=\\"regular-text\\" style=\\"color: rgb(38, 112, 255); padding: 0px; margin-top: -8px;\\" type=\\"button\\" class=\\"btn btn-link\\">Send feedback</button></div><div style=\\"padding: 80px 0px 20px 0px;\\" class=\\"col-2\\"><a id=\\"regular-text\\" style=\\"color: rgb(38, 112, 255);\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"https://github.com/Covaid-co/covaid-web\\">Github</a><br><a id=\\"regular-text\\" style=\\"color: rgb(38, 112, 255);\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"https://www.facebook.com/covaidco\\">Facebook</a><br><a id=\\"regular-text\\" style=\\"color: rgb(38, 112, 255);\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"https://www.instagram.com/covaidmutualaid/\\">Instagram</a><br></div><div style=\\"padding: 37px 20px 20px 100px;\\" class=\\"col-4\\"><p id=\\"regular-text\\" style=\\"color: rgb(123, 123, 123); font-weight: bold;\\">Current Partners</p><a id=\\"regular-text\\" style=\\"margin-bottom: 0px;\\" href=\\"https://www.facebook.com/bmoremutualaid/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">Baltimore Mutual Aid</a><br><a id=\\"regular-text\\" style=\\"margin-bottom: 0px;\\" href=\\"https://www.pittsburghmutualaid.com/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">Pittsburgh Mutual Aid</a><br><a id=\\"regular-text\\" style=\\"margin-bottom: 0px;\\" href=\\"https://www.covid-gca.org/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">Greater Charlotte Area Mutual Aid</a><br></div><div style=\\"padding: 78px 20px 20px 20px;\\" class=\\"col-4\\"><a id=\\"regular-text\\" style=\\"margin-bottom: 0px;\\" href=\\"https://www.facebook.com/groups/200572921276575/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">Delaware Mutual Aid</a><br><a id=\\"regular-text\\" style=\\"margin-bottom: 0px;\\" href=\\"https://ccomcovid.wixsite.com/covid\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">CCOM COVID-19 Task Force</a><br><p id=\\"regular-text\\" style=\\"margin-bottom: 0px;\\">Indy COVID-19 Neighbor Response Team</p></div></div></div></footer></div><div class=\\"react-toast-notifications__container css-mrh2d-ToastContainer\\" aria-hidden=\\"true\\"></div>"`;
